src/chocolatey.resources/helpers/functions/Uninstall-ChocolateyPath.ps1

```diff
@@ -0,0 +1,117 @@
+# Copyright 2011 - Present RealDimensions Software, LLC & original authors/contributors from https://github.com/chocolatey/chocolatey
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function Uninstall-ChocolateyPath {
+<#
+.SYNOPSIS
+**NOTE:** Administrative Access Required when `-PathType 'Machine'.`
+
+This puts a directory to the PATH environment variable of the
+requested scope (Machine or User).
+
+.DESCRIPTION
+Removes path from target path scope.  Removes multiple occurances (if they exist)
+and all occurances with or without a trailing slash.
+
+.NOTES
+This command will assert UAC/Admin privileges on the machine if
+`-PathType 'Machine'`.
+
+This is used when the application/tool is not being linked by Chocolatey
+(not in the lib folder).
+
+.INPUTS
+None
+
+.OUTPUTS
+None
+
+.PARAMETER PathToUninstall
+The full path to a location to remove from the PATH.
+
+.PARAMETER PathType
+Which PATH to remove it from. If specifying `Machine`, this requires admin
+privileges to run correctly.
+
+.PARAMETER IgnoredArguments
+Allows splatting with arguments that do not apply. Do not use directly.
+
+.EXAMPLE
+Uninstall-ChocolateyPath -PathToUninstall "$($env:SystemDrive)\tools\gittfs"
+
+.EXAMPLE
+Uninstall-ChocolateyPath "$($env:SystemDrive)\Program Files\MySQL\MySQL Server 5.5\bin" -PathType 'Machine'
+
+.LINK
+Install-ChocolateyPath
+
+.LINK
+Install-ChocolateyEnvironmentVariable
+
+.LINK
+Get-EnvironmentVariable
+
+.LINK
+Set-EnvironmentVariable
+
+.LINK
+Get-ToolsLocation
+#>
+param(
+  [parameter(Mandatory=$true, Position=0)][alias("Path")][string] $PathToUninstall,
+  [parameter(Mandatory=$false, Position=1)][ValidateSet('User','Machine')][System.EnvironmentVariableTarget] $pathType = [System.EnvironmentVariableTarget]::User,
+  [parameter(ValueFromRemainingArguments = $true)][Object[]] $ignoredArguments
+)
+  Write-Debug "Running 'Uninstall-ChocolateyPath' with PathToUninstall:`'$PathToUninstall`'";
+  Write-Output "Only evaluating and updating path scope `"$pathType`", path will not be assessed nor removed for other scope, so path may exist in other scope as well."
```

Comment:
I know it is widely believed that `Write-Host` is wrong, and it is under most circumstances. When it comes to logging, we prefer `Write-Host` in the Chocolatey installer module. It doesn't cause issues when you want to return actual things like `Write-Output` does. So please convert these to `Write-Host`.

Yes, I am familiar with http://www.jsnover.com/blog/2013/12/07/write-host-considered-harmful/:

* The post is about scripts that will be used by other tools.
* `Write-Output` ends up in the pipeline (aka is part of the return value from the function). This can have (and has had) disastrous results, so we prefer `Write-Host`. I prefer real life examples to rhetoric, so https://github.com/chocolatey/choco/commit/44306efbe55b982d92c9c23ba3b44df01e6f242e
* We control the hosts this runs under.
* We simply want logging messages.
* This is not a library or framework, but an end tool, which doesn't appear to be considered in the post.

Consider `Write-Output` == `Log.Info()` and `Write-Host` == `Log.Info()` for the Chocolatey framework.
Yes, I realize this means we should have probably set up a logging utility. We could go on all day long about how we should have brought in an actual logging facility like log4net, but honestly at the end of the day it is added work to provide what `Write-Host` already provides.